src/graphics/engine/text.cpp

```diff
@@ -780,32 +874,46 @@ void CText::DrawString(const std::string &text, std::vector<FontMetaChar>::itera
 
 void CText::StringToUTFCharList(const std::string &text, std::vector<UTF8Char> &chars)
 {
-    unsigned int index = 0;
-    unsigned int totalLength = text.length();
+    std::size_t index       = 0;
+    std::size_t totalLength = text.length();
     while (index < totalLength)
     {
         UTF8Char ch;
 
-        int len = StrUtils::Utf8CharSizeAt(text, index);
+        short len = StrUtils::Utf8CharSizeAt(text, index);
+        if (len == 0)   //warning & skip all : Risk of infinite loop !!
+        {
+            GetLogger()->Warn("Risk of infinite loop 6 -CText::StringToUTFCharList !!\n");
```

Comment:
This code with warnings is repeating all the time, how about extracting it to a function?